src/container-mapping.ts

```diff
@@ -0,0 +1,184 @@
+            listeners: {
+                stdout: (data: Buffer) => {
+                    reportData.dockerEvents = reportData.dockerEvents.concat(data.toString());
+                }
+            }
+        });
+        await exec.exec('docker images --format CreatedAt={{.CreatedAt}}::Repo={{.Repository}}::Tag={{.Tag}}::Digest={{.Digest}}', null, {
+            listeners: {
+                stdout: (data: Buffer) => {
+                    reportData.dockerImages = reportData.dockerImages.concat(data.toString());
+                }
+            }
+        });
+
+        core.debug("Finished data collection, starting API calls.");
+
+        await this.sendReport(reportData, sendReportRetryCount);
+    }
+
+    /**
+     * Sends a report to Defender for DevOps and retries on the specified count
+     * @param data the data to send
+     * @param retryCount the number of time to retry
+     * @returns a Promise
+     */
+    private async sendReport(data: Object, retryCount: number = 0): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            do {
+                try {
+                    await this._sendReport(data);
+                    resolve();
+                    break;
+                } catch (error) {
+                    if (retryCount == 0) {
+                        reject('Failed to send report: ' + error);
+                    } else {
+                        retryCount--;
+                        core.debug(`Retrying API call. Retry count: ${retryCount}`);
+                    }
+                }
+            } while (retryCount >= 0)
+        });
+    }
+    
+    /**
+     * Sends a report to Defender for DevOps
+     * @param data the data to send
+     * @returns a Promise
+     */
+    private async _sendReport(data: Object): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            let apiTime = new Date().getMilliseconds();
+            var bearerToken = await core.getIDToken();
+            let url: string = "https://dfdinfra-afdendpoint-dogfood-dqgpa4gjagh0arcw.z01.azurefd.net/github/v1/container-mappings";
+            let options = {
+                method: 'POST',
+                timeout: 2500,
+                headers: {
+                    'Content-Type': 'application/json',
+                    'Authorization': 'Bearer ' + bearerToken
+                },
+                data: data
+            };
+            core.debug(`${options['method'].toUpperCase()} ${url}`);
+    
+            const req = https.request(url, options, (res) => {
+                let resData = '';
+                res.on('data', (chunk) => {
+                    resData += chunk.toString();
+                });
+    
+                res.on('end', () => {
+                    core.debug('API calls finished. Time taken: ' + (new Date().getMilliseconds() - apiTime) + "ms");
+                    core.debug('Response: ' + resData);

```

Comment:
This is outputting empty data - https://github.com/davidknise/security-devops-action/actions/runs/6577755196/job/17870065781#step:12:14

> API calls finished. Time taken: 137ms
  Response: 

It would be great if we can add response code or something here for easy debugging (down the line we can think of downloading this log file in the backend service to get more debug data on why a push failed or something)